dsn/dsn.go

```diff
@@ -0,0 +1,39 @@
+// Package dsn meant to work with Delivery Status Notification (DSN) per rfc3464:
+// https://datatracker.ietf.org/doc/html/rfc3464
+package dsn
+
+import (
+	"net/textproto"
+	"strings"
+)
+
+// Report represents delivery status report as per https://datatracker.ietf.org/doc/html/rfc6522.
+// It is container of the MIME type multipart/report and contains 3 parts.
+type Report struct {
+	// Explanation contains a human-readable description of the condition(s) that caused the report to be generated.
+	Explanation Explanation
+	// DeliveryStatus provides a machine-readable description of the condition(s) that caused the report to be generated.
+	DeliveryStatus DeliveryStatus
+	// OriginalMessage is optional original message or its portion.
+	OriginalMessage []byte
```

Comment:
should we have here `[]byte` or `Envelope`?

Reply:
I think it's good to leave as bytes for now, not worth paying CPU cycles for a parse unless the user needs it.  Plus with the new decoding options, we'd probably want them to choose how to parse it.